livekit-plugins/livekit-plugins-azure/livekit/plugins/azure/stt.py

```diff
@@ -127,44 +137,57 @@ def __init__(
+        if language:
+            self._opts.languages = [language]
+            self._reconnect_event.set()
 
     async def _run(self) -> None:
-        self._stream = speechsdk.audio.PushAudioInputStream(
-            stream_format=speechsdk.audio.AudioStreamFormat(
-                samples_per_second=self._opts.sample_rate,
-                bits_per_sample=16,
-                channels=self._opts.num_channels,
+        while True:
+            self._stream = speechsdk.audio.PushAudioInputStream(
+                stream_format=speechsdk.audio.AudioStreamFormat(
+                    samples_per_second=self._opts.sample_rate,
+                    bits_per_sample=16,
+                    channels=self._opts.num_channels,
+                )
             )
-        )
-        self._recognizer = _create_speech_recognizer(
-            config=self._opts, stream=self._stream
-        )
-        self._recognizer.recognizing.connect(self._on_recognizing)
-        self._recognizer.recognized.connect(self._on_recognized)
-        self._recognizer.speech_start_detected.connect(self._on_speech_start)
-        self._recognizer.speech_end_detected.connect(self._on_speech_end)
-        self._recognizer.session_started.connect(self._on_session_started)
-        self._recognizer.session_stopped.connect(self._on_session_stopped)
-        self._recognizer.start_continuous_recognition()
-
-        try:
-            await asyncio.wait_for(
-                self._session_started_event.wait(), self._conn_options.timeout
+            self._recognizer = _create_speech_recognizer(
+                config=self._opts, stream=self._stream
             )
-
-            async for input in self._input_ch:
-                if isinstance(input, rtc.AudioFrame):
-                    self._stream.write(input.data.tobytes())
-
-            self._stream.close()
-            await self._session_stopped_event.wait()
-        finally:
-
-            def _cleanup():
-                self._recognizer.stop_continuous_recognition()
-                del self._recognizer
-
-            await asyncio.to_thread(_cleanup)
+            self._recognizer.recognizing.connect(self._on_recognizing)
+            self._recognizer.recognized.connect(self._on_recognized)
+            self._recognizer.speech_start_detected.connect(self._on_speech_start)
+            self._recognizer.speech_end_detected.connect(self._on_speech_end)
+            self._recognizer.session_started.connect(self._on_session_started)
+            self._recognizer.session_stopped.connect(self._on_session_stopped)
+            self._recognizer.start_continuous_recognition()
+
+            try:
+                await asyncio.wait_for(
+                    self._session_started_event.wait(), self._conn_options.timeout
+                )
+
+                async for input in self._input_ch:
+                    if self._reconnect_event.is_set():
+                        break
```

Comment:
This isn't ideal, we require to have any input to start the reconnection. We should only wait for `self._reconnect_event` and not `self._reconnect_event + self._input.ch.__anext__`